### ISSUE TITLE
Summarize your request. (e.g., "Request to add ROS to Open3D PointCloud conversion support?")

### ISSUE THREAD

**Issue by Ostyk** [isl-org/Open3D#5563]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Proposed new feature or change

Please include the ability to include setting distortion coefficients. 
I have seen that many have asked for this feature to be included. Personally, the problem I'm facing is that when trying to 3d points from a world view to a 2d depth map using `open3d.visualization.rendering.OffscreenRenderer.render_to_depth_image` there is no way to include the distortion coefficients in the camera model (http://www.open3d.org/docs/release/python_api/open3d.camera.html). Kindly advise on how to go about this problem with existing methods. 

Error is on the edges of the image. I have used this as it's an enormous speed up to `openCV projectpoints` but if I cannot include the camera coeffs then this is widely inaccurate. 

### References

Basically a copy of https://forum.open3d.org/t/open3d-geometry-pointcloud-create-from-rgbd-image-needs-to-consider-lense-distortion/466 which has not been answered since April 2021.
